housekeeper/store/cli.py

```diff
@@ -209,14 +209,34 @@ def delete_case(context, yes, case_name):
 @click.option('-f', '--force', is_flag=True,
               help='auto confirm and override sanity checks')
 @click.option('-d', '--date', help="date of the particular run")
-@click.argument('case_name')
+@click.option('-a', '--all', 'clean_all', is_flag=True, default=False, help="select all runs")
+@click.option('-b', '--before', help='runs before a date')
+@click.option('-a', '--after', help='runs after a date')
+@click.option('-u', '--untagged-only', is_flag=True, default=False, help='only remove assets without archive type')
+@click.argument('case_name', required=False)
 @click.pass_context
-def clean(context, force, date, case_name):
-    """Clean up files for an analysis."""
+def clean(context, force, date, clean_all, before, after, untagged_only, case_name):
+    """Clean up files for an analysis.
+    Providing a CASE_NAME will ignore --before, --after options
+    """
     manager = api.manager(context.obj['database'])
-    run_obj = run_orabort(context, case_name, date)
-    if force or click.confirm('Are you sure?'):
-        api.clean_up(context.obj['root'], run_obj, force=force)
+    if case_name:
+        run_objs = [run_orabort(context, case_name, date)]
+    else:
+        before_date = parse_date(before) if before else None
+        after_date = parse_date(after) if after else None
+        run_objs = api.runs(before=before_date, after=after_date)
+        if run_objs.first() is None:
+            log.error("no runs found")
+            context.abort()
+
+    if force or click.confirm('Cleaning following runs: {}'.
+                              format([ run_obj.case.name for run_obj in run_objs ])):
+        if untagged_only:
+            force = True
```

Comment:
For most runs we are going to clean this way, they won't be eligible for cleaning. And you want to get a confirmation question when cleaning multiple cases. So, setting force to `True` seemed like a hacky way of achieving that.